### PR TITLE
[ENH] Validation loss for deep learning clustering #3039

### PR DESCRIPTION
**Pull request by PipaFlores**
#### Reference Issues/PRs
Implements the enhances discussed in #3039

#### What does this implement/fix? Explain your changes.

it adds the `validation_split` arguement (default to 0) and makes splits for validation.
It also modifies the callbacks to monitor the validation loss, if provided.

This modifications are what I am using for my own research, I've tried to maintain the current style in the code. It seems functional in my own tests. However, It does NOT include a test module. If more clarity can be provided and the work is concise I might include it. 

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### Any other comments?
I would put special attention to the modifications in `_ae_resnet.py` as it has a custom training loop for `loss = "mul_rec"`
So the modifications are a bit more extensive.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thank you for this. Could you also include the ae_fcn and ae_bgru in your changes in order to stay consistent in terms of functionality ?

Also it seems the code checks everytime if self.validation_split not if self.validation_split > 0. I know that "0" is considered as False but it would be better if its clear that the condition is >0

The functionality should be tested with a simple test similar to existing ones [here](https://github.com/aeon-toolkit/aeon/blob/main/aeon/clustering/deep_learning/tests/test_deep_clusterer_features.py) where the new test should be added

**Comment by PipaFlores** [comment]
Done

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks for all the chanes ! one simple annoying comment, it seems for AEFCN and AEBGRU the new parameter is not in the docstring, all other + test looks great, could you just add the docstring for the parameter in these 2 models so we can merge ?

**Comment by PipaFlores** [comment]
Ups yea. There

**Review by hadifawaz1999 — APPROVED** [review]
Thanks LGTM!
